pkg/resources/function_commons.go

```diff
@@ -0,0 +1,341 @@
+package resources
+
+import (
+	"fmt"
+	"slices"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/schemas"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func init() {
+	javaFunctionSchema = setUpFunctionSchema(javaFunctionSchemaDefinition)
+	javascriptFunctionSchema = setUpFunctionSchema(javascriptFunctionSchemaDefinition)
+	pythonFunctionSchema = setUpFunctionSchema(pythonFunctionSchemaDefinition)
+	scalaFunctionSchema = setUpFunctionSchema(scalaFunctionSchemaDefinition)
+	sqlFunctionSchema = setUpFunctionSchema(sqlFunctionSchemaDefinition)
+}
+
+type functionSchemaDef struct {
+	additionalArguments           []string
+	functionDefinitionDescription string
+	runtimeVersionRequired        bool
+	runtimeVersionDescription     string
+	importsDescription            string
+	packagesDescription           string
+	handlerDescription            string
+	targetPathDescription         string
+}
+
+func setUpFunctionSchema(definition functionSchemaDef) map[string]*schema.Schema {
+	currentSchema := make(map[string]*schema.Schema)
+	for k, v := range functionBaseSchema {
+		v := v
+		if slices.Contains(definition.additionalArguments, k) || slices.Contains(commonFunctionArguments, k) {
+			currentSchema[k] = &v
+		}
+	}
+	if v, ok := currentSchema["function_definition"]; ok && v != nil {
+		v.Description = definition.functionDefinitionDescription
+	}
+	if v, ok := currentSchema["runtime_version"]; ok && v != nil {
+		if definition.runtimeVersionRequired {
+			v.Optional = false
+			v.Required = true
```

Comment:
If you are setting both here maybe `else` should also have both just in case?

Reply:
I was setting it here because doc generation was failing, I though it have to be set but then discovered that I was setting the Optional in the base schema 🤦 and forgot to remove it here. I will remove it now.